flax/struct.py

```diff
@@ -133,3 +135,20 @@ def from_state_dict(x, state):
 
 def field(pytree_node=True, **kwargs):
   return dataclasses.field(metadata={'pytree_node': pytree_node}, **kwargs)
+
+
+TNode = TypeVar('TNode', bound='PyTreeNode')
+
+
+class PyTreeNode():
```

Comment:
Could you explain the intuition behind this class?

Reply:
added a docstring